app/utils/commands.js

```diff
@@ -0,0 +1,241 @@
+const { UntilMorning } = require('./untilMorning')
+
+const allOptions = {
+  title: {
+    long: '--title',
+    short: '-T',
+    description: 'Specify title for next break (Long or Mini)',
+    withValue: true
+  },
+  text: {
+    long: '--text',
+    short: '-t',
+    description: 'Specify text for next break (Long Break only)',
+    withValue: true
+  },
+  skip: {
+    long: '--skip',
+    short: '-s',
+    description: 'Skip directly to this break (Long or Mini)',
+    withValue: false
+  },
+  duration: {
+    long: '--duration',
+    short: '-d',
+    description: 'Specify duration for pausing breaks (Pause only) [infinitely|until-morning|seconds (as number)]',
+    withValue: true
+  }
+}
+
+const allCommands = {
+  help: {
+    description: 'Show this help message'
+  },
+  version: {
+    description: 'Show current stretchly version'
+  },
+  reset: {
+    description: 'Reset breaks'
+  },
+  pause: {
+    description: 'Pause breaks',
+    options: [allOptions.duration]
+  },
+  resume: {
+    description: 'Resume from a pause'
+  },
+  toggle: {
+    description: 'Toggle breaks between resume/paused'
+  },
+  mini: {
+    description: 'Skips to next Mini Break',
+    options: [allOptions.title, allOptions.skip]
+  },
+  long: {
+    description: 'Skips to next Long Break',
+    options: [allOptions.text, allOptions.title, allOptions.skip]
+  }
+}
+
+const allExamples = [{
+  cmd: 'stretchly pause',
+  description: 'Pause breaks indefinitly'
+},
+{
+  cmd: 'stretchly pause -d 3600',
+  description: 'Pause breaks for one hour'
+},
+{
+  cmd: 'stretchly mini -T "Stretch up !" --skip',
+  description: 'Skips to next Mini Break with "Stretch up!" title'
+},
+{
+  cmd: 'stretchly long -T "Stretch up !" -t "Go stretch !"',
+  description: 'Set next long break title to "Stretch up !" and text to "Go stretch !"'
+}]
+
+// Parse cmd line, check if valid and put variables in a dedicated object
+class Command {
+  constructor (input, version) {
+    this.version = version
+    this.supported = allCommands
+
+    this.parse(input)
+  }
+
+  parse (input) {
+    this.command = input[0]
+
+    if (!this.supported[this.command]) {
+      throw new Error(`Error: command ${this.command} is not supported`)
+    }
+
+    this.options = this.getOpts(input.slice(1))
+  }
+
+  getOpts (opts) {
+    var options = {}
+
+    if (!this.supported[this.command].options) {
+      return null
+    }
+
+    for (let i = 0; i < opts.length; i++) {
+      const name = opts[i]
+      var valid = false
+
+      this.supported[this.command].options.forEach(opt => {
+        if (opt.long === name || opt.short === name) {
+          valid = true
+          if (opt.withValue) {
+            options[opt.long.slice(2)] = opts[i + 1]
+            i++
+          } else {
+            options[opt.long.slice(2)] = true
+          }
+        }
+      })
+
+      if (!valid) {
+        throw new Error(`Error: options ${name} is not valid for command ${this.command}`)
```

Comment:
Do we need to throw here? I don't think we should get window from command line 

![image](https://user-images.githubusercontent.com/4745859/95010888-2771da80-062d-11eb-8510-2c9b822629e7.png)

Reply:
It's something I wanted to discuss with you, I left this on purpose in order to get a clear error. Since it's in the second instance, there is no way to get clear feedback. I can log it to stdout in the main instance. Another solution would be a notification, maybe ?

Reply:
I replaced it with a log, see https://github.com/papey/stretchly/commit/554a61293032a69859973f50b6e5c52be373b4b2

Reply:
I guess we should try to do as much of validation as possible in command instance, with errors to output. So that primary instance doesn't need to care. 

I am not sure for example what `checkInForward` is for in main.

Reply:
I renamed it and add some comments, it's the function run in the main instance to return early if the command is a `second-instance` kind (`version` and `help` atm).

Reply:
I fixed it naively here : https://github.com/hovancik/stretchly/pull/700/commits/42c3b2e440645432d68ea710d5d07abf38dadb4e